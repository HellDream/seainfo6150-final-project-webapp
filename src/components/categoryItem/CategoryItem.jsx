import React from 'react';
import styles from './CategoryItem.module.css';
import { Link } from 'react-router-dom';
const CategoryItem = (props) => {
    return (
        <li className={styles.category}>
            <div className={styles.imageBlock}>
                <img
                    className={styles.image}
                    src={props.category.imageURI}
                    alt={`${props.title}`}
                />
            </div>
            <Link to={props.category.title} className={styles.title}>
                {props.category.title}
            </Link>
        </li>
    );
};

export default CategoryItem;
