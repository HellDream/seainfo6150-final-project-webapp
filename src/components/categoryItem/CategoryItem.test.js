import React from 'react';
import CategoryItem from './CategoryItem';
import { render } from '@testing-library/react';

import img from '../../images/books.jpg';
import { BrowserRouter } from 'react-router-dom';
const category = {
    slug: 1,
    imageURI: img,
    title: 'books',
};
describe('CategoryItem tests', () => {
    it('renders correctly', () => {
        const { container } = render(
            <BrowserRouter>
                <CategoryItem category={category} />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });
});
