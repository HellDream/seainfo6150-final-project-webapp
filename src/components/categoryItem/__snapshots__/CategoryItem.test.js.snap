// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem tests renders correctly 1`] = `
<div>
  <li
    class="category"
  >
    <a
      class="link"
      href="/books"
    >
      <div
        class="imageBlock"
      >
        <img
          alt="books"
          class="image"
          src="books.jpg"
        />
      </div>
      <h3
        class="title"
      >
        books
      </h3>
    </a>
  </li>
</div>
`;
